src/formula/variant_private.hpp

```diff
@@ -54,6 +54,9 @@ class variant_value_base;
 
 using value_base_ptr = std::shared_ptr<variant_value_base>;
 
+/**
+ * Helper functions to cast a @ref variant_value_base to a new derived type.
```

Comment:
You can't use one doc comment for two functions like this. Make "function" singular and duplicate the comment for each function.

If the duplication _really_ bugs you though, there _is_ a [`@copydoc` command](https://www.stack.nl/~dimitri/doxygen/manual/commands.html#cmdcopydoc), but that might be a little difficult to use with a template function.